File: recursion_func/factorial.py
# Factorial Algorithm

# using for - repeat
def factorial_repeat(num):

    result = 1
    
    for i in reversed(range(1, num + 1)):
        result = result * i

    return result

# using recursion function
def factorial_recursion(num):

    if num == 1:
        return 1
    
    return num * factorial_recursion(num - 1)
    

print(factorial_repeat(5))
print(factorial_recursion(5))

File: recursion_func/factorial.c
// Factorial Algorithm

#include <stdio.h>

// using for - repeat
int factorial_repeat(int num) {

	int result = 1;

	for(int i = num; i > 0; --i) {
		result = result * i;
	}

	return result;
}

// using recursion function
int factorial_recursion(int num) {

	if(num == 1) {
		return 1;
	}

	return num * factorial_recursion(num - 1);
}

int main() {
	printf("Factorial using for-repeat = %d \n", factorial_repeat(5));
	printf("Factorial using recursion = %d \n", factorial_recursion(5));
}
